File: src/VBNetFx/FAC_Generic/frmLogin.vb
Imports System.Windows.Forms

<CLSCompliant(False)> _
Public Class frmLogin

    Dim myLoginInfo As UserInfo

    Public Function Login(ByVal Subsidiaries As SubsidiaryInfoCollection, ByVal PreselectSSID As Integer, ByVal locLoginHistory As LoginHistory) As UserInfo
        Using Me
            Dim locCount As Integer = 0
            Dim locPreselectIndex As Integer = 0
            For Each locSubsidiary As SubsidiaryInfo In Subsidiaries
                If PreselectSSID > 0 Then
                    If locSubsidiary.IDSubsidiary = PreselectSSID Then
                        locPreselectIndex = locCount
                    End If
                End If
                cmbSubsidiary.Items.Add(locSubsidiary)
            Next
            If cmbSubsidiary.Items.Count > 0 Then
                cmbSubsidiary.SelectedIndex = locPreselectIndex
            End If
            For Each locString As String In locLoginHistory
                cmbUsernames.Items.Add(locString)
            Next
            cmbUsernames.Text = locLoginHistory.LastLoginName
            Me.ShowDialog()
            If Me.DialogResult = Windows.Forms.DialogResult.OK Then
                Return myLoginInfo
            Else
                Return Nothing
            End If
        End Using
    End Function

    Private Sub btnOK_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnOK.Click
        myLoginInfo = New UserInfo(CType(cmbSubsidiary.SelectedItem, SubsidiaryInfo).IDSubsidiary, _
                                   cmbUsernames.Text, txtPassword.Text, _
                                   FacessoGeneric.SQLConnectionString)

        If Not myLoginInfo.Authenticated Then
            MessageBox.Show(myLoginInfo.LoggedInFailedReason, "Fehler bei Login:", _
            MessageBoxButtons.OK, MessageBoxIcon.Exclamation)
            Return
        End If
        Me.Hide()
        Me.DialogResult = Windows.Forms.DialogResult.OK

    End Sub

    Private Sub btnCancel_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCancel.Click
        Me.DialogResult = Windows.Forms.DialogResult.Cancel
        Me.Hide()
    End Sub

    Private Sub txtPassword_TextChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles txtPassword.TextChanged
        If txtPassword.Text <> "" Then
            Me.AcceptButton = Me.btnOK
        Else
            Me.AcceptButton = Nothing
        End If
    End Sub

    Protected Overrides Sub OnLoad(ByVal e As System.EventArgs)
        MyBase.OnLoad(e)
        Me.Visible = True
        Application.DoEvents()
        If cmbUsernames.Text = "" Then
            cmbUsernames.Focus()
        Else
            txtPassword.Focus()
        End If
    End Sub

    Private Sub frmLogin_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

    End Sub
End Class

File: src/MixNetFx/FAC_Generic/frmLogin.cs
using System.Windows.Forms;

namespace Facesso
{
    [System.CLSCompliant(false)]
    public partial class frmLogin : Form
    {
        private UserInfo myLoginInfo;

        public frmLogin()
        {
            InitializeComponent();
        }

        public UserInfo Login(SubsidiaryInfoCollection subsidiaries, int preselectSSID, LoginHistory locLoginHistory)
        {
            using (this)
            {
                int locCount = 0;
                int locPreselectIndex = 0;
                foreach (SubsidiaryInfo locSubsidiary in subsidiaries)
                {
                    if (preselectSSID > 0)
                    {
                        if (locSubsidiary.IDSubsidiary == preselectSSID)
                            locPreselectIndex = locCount;
                    }

                    cmbSubsidiary.Items.Add(locSubsidiary);
                    locCount++;
                }
                if (cmbSubsidiary.Items.Count > 0)
                    cmbSubsidiary.SelectedIndex = locPreselectIndex;

                foreach (string locString in locLoginHistory)
                    cmbUsernames.Items.Add(locString);

                cmbUsernames.Text = locLoginHistory.LastLoginName;
                ShowDialog();

                if (DialogResult == System.Windows.Forms.DialogResult.OK)
                    return myLoginInfo;
                else
                    return null;
            }
        }

        private void btnOK_Click(object sender, System.EventArgs e)
        {
            myLoginInfo = new UserInfo(
                ((SubsidiaryInfo)cmbSubsidiary.SelectedItem).IDSubsidiary,
                cmbUsernames.Text,
                txtPassword.Text,
                FacessoGeneric.SQLConnectionString);

            if (!myLoginInfo.Authenticated)
            {
                MessageBox.Show(myLoginInfo.LoggedInFailedReason, "Fehler bei Login:",
                    MessageBoxButtons.OK, MessageBoxIcon.Exclamation);
                return;
            }
            Hide();
            DialogResult = System.Windows.Forms.DialogResult.OK;
        }

        private void btnCancel_Click(object sender, System.EventArgs e)
        {
            DialogResult = System.Windows.Forms.DialogResult.Cancel;
            Hide();
        }

        private void txtPassword_TextChanged(object sender, System.EventArgs e)
        {
            AcceptButton = txtPassword.Text != "" ? (IButtonControl)btnOK : null;
        }

        protected override void OnLoad(System.EventArgs e)
        {
            base.OnLoad(e);
            Visible = true;
            Application.DoEvents();
            if (cmbUsernames.Text == "")
                cmbUsernames.Focus();
            else
                txtPassword.Focus();
        }

        private void frmLogin_Load(object sender, System.EventArgs e) { }
    }
}
